src/tokenized-line.coffee

```diff
@@ -59,7 +74,7 @@ class TokenizedLine
     screenColumn = 0
     currentBufferColumn = 0
     for token in @tokens
-      break if currentBufferColumn > bufferColumn
+      break if currentBufferColumn + token.bufferDelta > bufferColumn
```

Comment:
Cool, that's what I was thinking too. I've been looking through the code and it looks like `screenPositionForBufferPosition` interacts more interestingly with soft tabs than with hard tabs. If you have a soft tab, then `screenPositionForBufferPosition` clips to the edges of it when called for any of the individual spaces making it up. Hard tabs only take up one buffer character, and `screenPositionForBufferPosition` calculates the tab width.
The specs are written to test around hard tabs, soft wraps, and folds, but not for soft tabs. I think those are the only tests we're missing.